ananse/influence.py

```diff
@@ -33,41 +33,165 @@
+    )  # sort based on probability score
+
+    # check method of selecting edges, if a interaction set is provided, export those from the GRN
+    # if not export the edges of the network.
+    if interactions == None:
+        rnet = rnet.head(edges)
+    else:
+        rnet = rnet[rnet.tf_target.isin(interactions)]
+
+    edge_atribute_length = len(
+        rnet.columns
+    )  # check if network is full output or simple output
     for _, row in rnet.iterrows():
         source, target = row[0].split("_", 1)
-        weight = 0 if len(row) < 2 else float(row[1])
-        try:
-            G.add_edge(source, target, weight=weight, n=1)
-        except Exception:
-            logger.error("Could not parse edge weight.")
-            raise
+        weight = 0 if len(row) < edge_atribute_length else float(row["prob"])
+        # when the GRN with the full output is generated, more data can be exported to the diffnetwork
+        # all the extra data is stored in edge attributes
+        if edge_atribute_length == 6:
+            tf_expression = (
+                0 if len(row) < edge_atribute_length else float(row["tf_expression"])
+            )
+            target_expression = (
+                0
+                if len(row) < edge_atribute_length
+                else float(row["target_expression"])
+            )
+            weighted_binding = (
+                0 if len(row) < edge_atribute_length else float(row["weighted_binding"])
+            )
+            activity = 0 if len(row) < edge_atribute_length else float(row["activity"])
+            try:
+                G.add_edge(
+                    source,
+                    target,
+                    weight=weight,
+                    weighted_binding=weighted_binding,
+                    TF_expression=tf_expression,
+                    TF_activity=activity,
+                    TG_expression=target_expression,
+                )
+            except Exception:
+                print("Could not parse edge weight.")
+                raise
+
+        # if there is only prob score, only that is inc in the GRN
+        else:
+            try:
+                G.add_edge(source, target, weight=weight, n=1)
+            except Exception:
+                print("Could not parse edge weight.")
+                raise
     return G
 
 
-def difference(S, R):
-    """Calculate the network different between two cell types."""
-    DIF = nx.create_empty_copy(R)
-    for (u, v, d) in S.edges(data=True):
-        if (u, v) not in R.edges:
-            DIF.add_edge(u, v, weight=d["weight"], n=1)
-        else:
-            diff_weight = S.edges[u, v]["weight"] - R.edges[u, v]["weight"]
-            if diff_weight > 0:
+def difference(GRN_source, GRN_target):
+    """Calculate the network different between two GRNs.
+    It first takes the nodes from both networks, and subsequently
+    First adds the edges from the target network that are missing in
```

Comment:
2 starting sentences?

other than that, THANK YOU for adding proper docstrings!!!